tests/core/test_schema_components.py

```diff
@@ -179,6 +179,23 @@ def test_multi_index_index():
         schema.validate(df_fail)
 
 
+def test_single_index_multi_index_mismatch():
+    """Tests the failure case that attempting to validate a MultiIndex DataFrame
+    against a single index schema raises a SchemaError with a constructive error
+    message."""
+    ind = pd.MultiIndex.from_tuples(
+        [("a", "b"), ("c", "d"), ("e", "f")],
+        names=("one", "two"),
+    )
+    df_fail = pd.DataFrame(columns=["alpha", "beta"], index=ind)
+    schema = DataFrameSchema(index=Index(name="key"))
+
+    if isinstance(df_fail.index, pd.MultiIndex) and isinstance(
```

Comment:
`pytest.raises` should be used as a context manager:

```python
with pytest.raises(errors.SchemaError):
    schema.validate(df_fail)
```

Note that you don't need the conditional statement since the df and schema you've defined already fulfill those properties.